odilia/src/tower/unwrap_svc.rs

```diff
@@ -0,0 +1,46 @@
+use futures::FutureExt;
+use std::convert::Infallible;
+use std::future::Future;
+use std::marker::PhantomData;
+use std::task::{Context, Poll};
+use tower::Service;
+
+pub struct UnwrapService<S, Req, Res, R, E, F> {
```

Comment:
this service could, probably, have been done with the primitive which makes services from cloasures, service::from_fn or something similar. Perhaps do that, if possible?

Reply:
Ah... nameable types are fun.

The issue here is that I would have to name the type if I used `service_fn` inside our `ServiceExt` trait. I tried that route: it leads to madness! The compiler will insist: "expected associated type F, found type {closure@main.rs:123} ..." but obviously that is not the case.

My rebuttal is that this is actually quite a trivial case of a good service/combinator set, and since the implementation is more-or-less trivial (and safe because we protect against the error being anything other than Infallible....) we have no reason to drop it "just because we might be able to emulate this with a `service_fn`, even though, as I just mentioned, this actually would not work.

Reply:
well, that error can be because the function, or closure in your case, doesn't implement the right Fn* trait. Maybe try with an actual function? I dk, I'm trying to come up with ways of making this code change not be extremely huge, also be simpler to reason about in the future, but we work with what we have and what  the compiler allows for now

Reply:
I'd rather a type: `UnwrapService`, than a `ServiceFn<ServiceFn<ServuceFn<...>>>`

Imagine debugging that! At least with our explicit type we get debuggability.